libsolidity/parsing/Parser.cpp

```diff
@@ -1056,6 +1056,28 @@ ASTPointer<Expression> Parser::parsePrimaryExpression()
 		expectToken(Token::RParen);
 		return nodeFactory.createNode<TupleExpression>(components);
 	}
+	case Token::LBrack:
+	{
+		// Inline array expression
+		// Special cases: [] is empty tuple type, (x) is not a real tuple, (x,) is one-dimensional tuple
```

Comment:
I think we do not have special cases for inline arrays, `[]` is just an empty array, `[x]` is a one-element array and especially `[x,]` is not valid, you cannot leave out elements.